spec/controllers/taxonomy_signups_controller_spec.rb

```diff
@@ -3,19 +3,65 @@
 RSpec.describe TaxonomySignupsController do
   include GdsApi::TestHelpers::ContentStore
 
-  describe "GET new" do
-    it 'redirects to root unless valid query param provided' do
-      get :new, bad_param: '/education/some-rando-item'
+  shared_examples 'handles bad input data correctly' do
+    it 'redirects to root if topic param is missing' do
+      make_request(bad_param: '/education/some-rando-item')
+
+      expect(response.status).to eq 302
+      expect(response.location).to eq 'http://test.host/'
+    end
+
+    it "redirects to root if topic param isn't a valid path" do
+      get :new, topic: '/with unencoded spaces'
+
+      expect(response.status).to eq 302
+      expect(response.location).to eq 'http://test.host/'
+    end
+
+    it "redirects to root if topic param isn't interpreted as a string" do
+      get :new, topic: ['/a']
 
       expect(response.status).to eq 302
       expect(response.location).to eq 'http://test.host/'
     end
 
     it 'errors if no taxon found' do
       content_store_does_not_have_item('/education/some-rando-item')
-      get :new, topic: '/education/some-rando-item'
+      make_request(topic: '/education/some-rando-item')
 
       expect(response.status).to eq 404
     end
+
+    it 'redirects to root unless the content item is a taxon' do
```

Comment:
This test is for the case where it isn't a taxon, though... There's a [cucumber feature for the happy path](https://github.com/alphagov/email-alert-frontend/blob/2b113232be7f1c82adcb3519d38b9a37917d82da/features/taxonomy_email_alerts.feature).